larpandoracontent/LArObjects/LArMCParticle.h

```diff
@@ -28,6 +28,8 @@ class LArMCParticleParameters : public object_creation::MCParticle::Parameters
 {
 public:
     pandora::InputInt   m_nuanceCode;               ///< The nuance code
+    pandora::InputBool  m_isDR;
+    pandora::InputBool  m_isDecay;
```

Comment:
(This comment applies across the entire file)

I think it's worthwhile to make this more generic, and I think this can be done with relatively little work. If you recall our previous discussions about the process information from some time ago, I started putting together such a feature branch (comparison to the 3.22.03 base here):

https://github.com/PandoraPFA/LArContent/compare/feature/larpandoracontent_v03_22_03...AndyChappell:feature/mc_updates

I've not had the opportunity to finish this off, but it should be pretty close to being release-worthy and I'd be happy to see it go out as part of this PR - essentially it just needs some versioning of the LArMCParticle so we can still load pndr files that don't have this information (my take would be this becomes the default version and old versions have to specify version 1 if they want it, but maybe @johnmarshall80 has a view on this?). Happy to chat about the versioning if needed.

I think you can largely replace what's here with the cherry picked commits from my feature branch, as the logic is essentially the same (The visualisation tools are unnecessary of course, but if it turns out to be easier to include them as part of the cherry pick, that's fine). If you want to retain the GetIsDR and GetIsDecay helper functions, I think that's fine, they can just perform a check against the value returned by GetProcess -  though I'd drop the "Get" from the function names.

Reply:
Hi @AndyChappell, Yes, I'd agree with the new version becoming the default because pndr files are for development use by the team. We should ensure that the XML snippets to switch the version of the LArMCParticle are documented somewhere, and we should take a look at the error handling when different file/deserialising combinations are used to check that the error messages aren't too cryptic! Nice suggestion to connect-up these features. Thanks!